File: src/CourseService/Application/Lessons/Queries/Dtos/LessonSummaryDto.cs
using Courses.Application.Actions.Primitives;
using Courses.Domain.Courses.Primitives;
using Courses.Domain.Lessons.Primitives;
using Courses.Domain.Shared.Primitives;

namespace Courses.Application.Lessons.Queries.Dtos;

public record LessonSummaryDto(
    CourseId CourseId,
    LessonId LessonId,
    Title Title,
    Description Description,
    int Index,
    TimeSpan? Duration,
    bool IsPreview,
    ImageUrl? ThumbnailUrl,
    IReadOnlyCollection<LessonAction> AllowedActions
);

File: Frontend/src/features/lessons/types/LessonSummaryDto.ts
/**
 * Backend DTO: Matches LessonSummaryDto from CourseService
 * Used for network layer communication
 */
export interface LessonSummaryDto {
  courseId: string;
  lessonId: string;
  title: string;
  description: string;
  index: number;
  duration: string | null;
  isPreview: boolean;
  thumbnailUrl: string | null;
}
